### PR TITLE
Make disconnected screen reason text scrollable

### PR DESCRIPTION
**Pull request by deirn**
Mostly happens when regsync failed caused by modpack update.

### CODE REVIEW

File: fabric-networking-api-v1/src/client/java/net/fabricmc/fabric/mixin/networking/client/DisconnectedScreenMixin.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			if (scroll < actualReasonHeight - reasonHeight) {
+				fillGradient(matrixStack, startX, y + reasonHeight - 10, endX, y + reasonHeight, 0, 0xFF000000);
+			}
+		}
+
+		return y + reasonHeight;
+	}
+
+	@Override
+	public void fabric_mouseScrolled(double mouseX, double mouseY, double amount) {
```
Comment (modmuss50) [inline fabric-networking-api-v1/src/client/java/net/fabricmc/fabric/mixin/networking/client/DisconnectedScreenMixin.java]: If im not mistake you should just able to override onMouseScrolled here, and remove the need for the 2nd mixin + interface?

Comment (modmuss50) [reply fabric-networking-api-v1/src/client/java/net/fabricmc/fabric/mixin/networking/client/DisconnectedScreenMixin.java]: Fixed my self as I had it cloned down for testing.

**Review by modmuss50 — APPROVED** [review]
Great, nice idea.

**Comment by Player3324** [comment]
The change is good, but maybe we should move it to some sort of utility/enhancement module? networking-api seems a bit inappropriate

**Comment by modmuss50** [comment]
Yeah.. We dont really have anything like this right now, could there be anything similar in the future?

**Comment by Player3324** [comment]
I believe so, e.g. around networking (e.g. error induced disconnect are a pain to investigate atm) but maybe also other areas. I'd go for a generic module for little enhancements, especially ones that are perceivable by the end user, but don't warrant explicit opt-in/out.

File: fabric-networking-api-v1/src/client/java/net/fabricmc/fabric/mixin/networking/client/DisconnectedScreenMixin.java (hunk 1 above)
Comment (deirn) [reply fabric-networking-api-v1/src/client/java/net/fabricmc/fabric/mixin/networking/client/DisconnectedScreenMixin.java]: Overriding `Screen`s method in the mixin is pretty much the same as `@Overwrite` so I avoided it.

Comment (modmuss50) [reply fabric-networking-api-v1/src/client/java/net/fabricmc/fabric/mixin/networking/client/DisconnectedScreenMixin.java]: Not really, as its adding a new method, not overriding anything. Can revert my change if you think its best?

**Review by Juuxel — APPROVED** [review]

**Comment by Technici4n** [comment]
Can't we keep this in networking API as long as the logic is not required anywhere else? It's impl-only, so we can freely move it around.

**Comment by Player3324** [comment]
For now maybe, would like to move it eventually though to not bundle optional "behavior" changes with an api.

**Review by Player3324 — APPROVED** [review]

**Pull request merged**